handler.go

```diff
@@ -283,8 +327,12 @@ func (p proxyHandler) handleRequest(ctx *Context, rw http.ResponseWriter, req *h
 		res.Close = true
 	}
 
-	writeResponse(rw, res)
-	return nil
+	// deal with 101 Switching Protocols responses: (WebSocket, h2c, etc)
+	if res.StatusCode == 101 {
```

Comment:
How about `http.StatusSwitchingProtocols`

Reply:
I left the integer values for compatibility with existing code.